include/TPP/Dialect/Tpp/TppOps.td

```diff
@@ -344,4 +391,111 @@ def Tpp_VNNIBrgemmOp : Tpp_Op<"vnni_brgemm"> {
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// FusedBrgemmOp
+//===----------------------------------------------------------------------===//
+
+def Tpp_FusedBrgemmOp : Tpp_Op<"fused_brgemm"> {
+  let summary = [{
+    Performs batch reduced matrix multiplication followed by bias addition and relu}];
+
+  let description = [{
+    The `tpp.fused_brgemm` is an implementation of the Batch GEMM operation in oneAPI.
+
+    Example:
+
+    ```mlir
+
+      tpp.fused_brgemm ins(%1: memref<3x6x4xf32>, %2: memref<3x4x4xf32>, %3: memref<4xf32>)
+                      out(%3: memref<6x4xf32>)
+    ```
+  }];
+
+  let arguments = (ins TppBRGEMMemrefInput:$batchMatrixA,
+                       TppBRGEMMemrefInput:$batchMatrixB,
+                       TppMemRef: $bias,
+		       TppMemRef:$matrixC);
+
+  let assemblyFormat = [{
+      `ins` `(` $batchMatrixA `:` type($batchMatrixA) `,`
+                $batchMatrixB `:` type($batchMatrixB) `,`
+		$bias `:` type($bias)  `)`
+      `out` `(` $matrixC `:` type($matrixC) `)` attr-dict
+  }];
+
+  let extraClassDeclaration = [{
+    MemRefType getMatrixCType() {
+      return getMatrixC().getType().cast<MemRefType>();
+    }
+
+    MemRefType getBatchMatrixAType() {
+      return getBatchMatrixA().getType().cast<MemRefType>();
+    }
+
+    MemRefType getBatchMatrixBType() {
+      return getBatchMatrixB().getType().cast<MemRefType>();
+    }
+  }];
+
+  let builders = [
+    OpBuilder<(ins "ValueRange":$inputs, "Value":$output)>
+  ];
+
+  let hasVerifier = 1;
+}
+
+//===----------------------------------------------------------------------===//
+// FusedVNNIBrgemmOp
+//===----------------------------------------------------------------------===//
+
+def Tpp_FusedVNNIBrgemmOp : Tpp_Op<"fused_vnni_brgemm"> {
```

Comment:
Perhaps an attribute for VNNI, too?

Reply:
Haven't added one, will track it as a part of the issues list to be taken up in the following week.